### ISSUE TITLE
Performance compare with `NetworkX`

### ISSUE THREAD

**Issue by acezen** [networkit/networkit#845]
hi, guys, The project is great. 
 Did you made a performance test or benchmark with NetworkX (especially algorithms) before?

**Comment by avdgrinten** [comment]
NetworKit should be some orders of magnitude faster for combinatorial algorithms (such as Betweenness, Closeness, community detection algorithms, distance algorithms, etc) since networkx implements them in pure Python (as far as I am aware). For algebraic algorithms such as PageRank or Eigenvalue centrality (which networkx implements on top of numpy+scipy), the performance difference might not be that great.

You'll need to test specific algorithms (and inputs) to get reliable performance results.

**Comment by angriman** [comment]
Here is an example of performance benchmarks of popular graph/network analysis frameworks including NetworKit and networkX: https://www.timlrx.com/blog/benchmark-of-popular-graph-network-packages

**Comment by hmeyerhenke** [comment]
Yes, we have (quite old) benchmark results for combinatorial algorithms that confirm what @avdgrinten said. Since then, comparisons to other packages also following the Python + C/C++ concept make more sense -- at least to us.
